### PR TITLE
PWGHF: Add prompt and feeddown histos to the Lc->pKpi task

### PR DESCRIPTION
**Pull request by DelloStritto**

### CODE REVIEW

**Comment by vkucera** [comment]
There are some conflicts.

**Comment by ginnocen** [comment]
Hi Luigi, is this still up to date?

**Comment by DelloStritto** [comment]
> Hi Luigi, is this still up to date?

Hi GianMichele, there are some conflicts. I will resolve them.

**Comment by vkucera** [comment]
@DelloStritto Any updates?

**Comment by DelloStritto** [comment]
> @DelloStritto Any updates?

Hi @vkucera . Sorry for the delay. I fixed the conflicts. Now everything should be fine.

File: Analysis/Tasks/PWGHF/taskLc.cxx [hunk 1]
```diff
@@ -108,8 +110,12 @@ struct TaskLcMC {
...
+     {"hPtRecSigNonPrompt", "3-prong candidates (matched, non-prompt);#it{p}_{T}^{rec.} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
      {"hPtRecBg", "3-prong candidates (rec. unmatched);#it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
      {"hPtGen", "3-prong candidates (gen. matched);#it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
+     {"hPtGenPrompt", "MC particles (matched, prompt);#it{p}_{T}^{gen.} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
+     {"hPtGenNonPrompt", "MC particles (matched, non-prompt);#it{p}_{T}^{gen.} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
      {"hPtGenSig", "3-prong candidates (rec. matched);#it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
      {"hCPARecSig", "3-prong candidates (rec. matched);cosine of pointing angle;entries", {HistType::kTH1F, {{110, -1.1, 1.1}}}},
      {"hCPARecBg", "3-prong candidates (rec. unmatched);cosine of pointing angle;entries", {HistType::kTH1F, {{110, -1.1, 1.1}}}},
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/taskLc.cxx]: Can you please adjust the histogram titles? (See `taskD0.cxx`)

File: Analysis/Tasks/PWGHF/taskLc.cxx [hunk 2]
```diff
@@ -140,7 +145,13 @@ struct TaskLcMC {
         auto indexMother = RecoDecay::getMother(particlesMC, candidate.index0_as<aod::BigTracksMC>().mcParticle_as<soa::Join<aod::McParticles, aod::HfCandProng3MCGen>>(), pdg::Code::kLambdaCPlus, true);
         auto particleMother = particlesMC.iteratorAt(indexMother);
         registry.fill(HIST("hPtGenSig"), particleMother.pt()); // gen. level pT
-        registry.fill(HIST("hPtRecSig"), candidate.pt());      // rec. level pT
+        auto ptRec = candidate.pt();
+        registry.fill(HIST("hPtRecSig"), ptRec); // rec. level pT
+        if (candidate.originMCRec() == Prompt) {
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/taskLc.cxx]: Please use explicit namespace qualifier: `OriginType::Prompt` instead of `Prompt`.

**Review by vkucera — CHANGES_REQUESTED** [review]

File: Analysis/Tasks/PWGHF/taskLc.cxx (hunk 1 above)
Comment (DelloStritto) [reply Analysis/Tasks/PWGHF/taskLc.cxx]: Done!

**Review by vkucera — DISMISSED** [review]
Thanks

**Review by vkucera — APPROVED** [review]

**Comment by vkucera** [comment]
@ginnocen Ready. Can you approve, please?

**Review by ginnocen — APPROVED** [review]

**Comment by vkucera** [comment]
@jgrosseo @iarsene All green and approved. Can you please merge?

**Review by jgrosseo — APPROVED** [review]

**Pull request merged**